test/re/frida/MethodTest.java

```diff
@@ -209,6 +210,39 @@ public void staticFieldCanBeRead() {
         assertEquals("" + Cipher.ENCRYPT_MODE, script.getNextMessage());
     }
 
+    @Test
+    public void replacementCanBeDoneOnReflection() {
+        loadScript("var classDef = Java.use('java.lang.Class');\n" +
+                   "var getMethod = classDef.getMethod.overload('java.lang.String',\n" +
+                                                               "'[Ljava.lang.Class;');\n" +
+                   "send('overload found');\n" + 
+
+                   "getMethod.implementation = function(name, array) {\n" +
+
+                       "var method = getMethod.call(this, name, array);\n" +
+                       "send(name + ' dereflected');" + 
+                       "return method;\n" +
+                   "}\n" +
+
+                   "send('implementation replaced');\n");
+
+        assertEquals("overload found", script.getNextMessage());
+        assertEquals("implementation replaced", script.getNextMessage());
+
+        Class reflector = Reflector.class;
+
+        java.lang.reflect.Method method;
+        try {
+            method = reflector.getMethod("reflected");
+
```

Comment:
I'd say this blank line should be dropped so the “action” and “observed effect” are together.

Reply:
ok